### PR TITLE
Add visionOS support

### PR DESCRIPTION
**Pull request by tom-matches**
### What and why?

Support building for native VisionOS
Based off the work from here - https://github.com/DataDog/dd-sdk-ios/pull/1330

### How?

- Extend places where other non iOS platforms are excluded

### Review checklist
- [ ] Feature or bugfix MUST have appropriate tests (unit, integration)
- [ ] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing changes

- I've run the lint script (`sh tools/lint/run-linter.sh`)
- Unit tests are passing locally for me

### Custom CI job configuration (optional)
- [ ] Run unit tests
- [ ] Run integration tests
- [ ] Run smoke tests

### CODE REVIEW

File: DatadogRUM/Sources/RUMVitals/VitalInfoSampler.swift [hunk 1]
```diff
@@ -41,12 +41,22 @@ internal final class VitalInfoSampler {
 
     private var timer: Timer?
 
+    private static var maximumFramesPerSecond: Double {
+        get {
+#if os(visionOS)
+            return 120.0
```
Comment (maciejburda) [inline DatadogRUM/Sources/RUMVitals/VitalInfoSampler.swift:47]: Would be great to add a comment explaining that it's hardcoded until visionOS is publicly available

**Review by maciejburda — APPROVED** [review]
LGTM

File: DatadogRUM/Sources/RUMVitals/VitalInfoSampler.swift (hunk 1 above)
Comment (tom-matches) [reply DatadogRUM/Sources/RUMVitals/VitalInfoSampler.swift:47]: Sorry I didn't get around to doing this, I can create another PR with a comment if you wish.

**Comment by johnHackworth** [comment]
Hey @tom-matches 👋 Even though this was reverted at the time, your push for visionOS support was heard — full official visionOS support has now shipped in **[v3.10.0](https://github.com/DataDog/dd-sdk-ios/releases/tag/3.10.0)**! RUM, Crash Reporting, Logs, Trace, Profiling, and Web View Tracking are all supported now 🎉 Thanks for the early contribution!

**Pull request merged**